### PR TITLE
Add bitsPerSample resource from taglib if available

### LINKED ISSUE #978: Add bits_per_sample (sample width)  to obj.res
Is there a way to make this resource information available for scripting? I would like to organise my audio files by format, like 
**FLAC | 24 bit | 96 KHz | stereo**. Taglib presents this as bitsPerSample if I'm not mistaken.

### PR DESCRIPTION
**Pull request by KarlStraussberger**
fixes #978 : It's available as `R_BITS_PER_SAMPLE`

Also add properties of additional resources (e.g. fanart) to script context `obj.res`

### CODE REVIEW

File: src/metadata/taglib_handler.cc [hunk 1]
```diff
@@ -520,6 +520,13 @@ void TagLibHandler::extractASF(TagLib::IOStream* roStream, const std::shared_ptr
     }
     populateGenericTags(item, asf);
 
+    auto audioProps = asf.audioProperties();
```
Comment (whyman) [inline src/metadata/taglib_handler.cc:523]: There is already handling for audioProperties in `populateGenericTags`, should be able to add it there.

Comment (KarlStraussberger) [reply src/metadata/taglib_handler.cc:523]: `TagLib::File` does not have the required audioProperties. The base method returns a simpler type with fewer methods.

**Comment by ghlupe2** [comment]
That does not seem to work. On the web UI I could not find a single flac file where "Edit Item, Show Details, bitsPerSample" is not empty. The reason might be that some file types like mp3 do NOT contain this information by design. I guess that taglib provides this information only on file type basis, but not overall.

**Comment by KarlStraussberger** [comment]
That's why the code looks duplicated. If there is no value, there are two options
- the file does not contain it - check it with tagreader
- taglib does not properly read it - what's your taglib version?

**Comment by ghlupe2** [comment]
Thanks for looking into this, I don't think that tagreader can read bitsPerSample, it is not covered by audioproperties.cpp, but exiftool does for all flac files I tested. My taglib version is the current one from github (1.12).

**Comment by ghlupe2** [comment]
I assume that in your environment this feature works well? 
I double checked against install_manifest.txt that all taglib 1.11.1 related files have been removed before 1.12 install. And taglib 1.11.1 related issue #1066 is fixed now, so I'm pretty sure that I am on a current Gerbera (v1.8.2-166-gc8e2bbe0) and Taglib version.

**Comment by KarlStraussberger** [comment]
Proof of the working feature at my server:
grafik

Did you reimport your library? Metadata are not updated without rescanning, i.e. stop gerbera, delete your sqlite-db file and start gerbera again.

**Comment by ghlupe2** [comment]
Great! This is an example of a file I just imported:
Screenshot 2021-07-19 11 49 17
Are more libraries or software to be considered in this process? My list of resources looks quite different (colorDepth, dc:language, ...)

**Comment by KarlStraussberger** [comment]
Everything should be done by taglib.

From the empty entries I can see that your running the first version using grb_cds_resource. Please update to the current head (like always, some flaws made it into the main branch). Maybe that solves the issue.

**Comment by ghlupe2** [comment]
I am on an almost current version but still see strange effects regarding this issue. Flac files with integrated artwork show the bits per sample information in Gebera Web UI, all the others don't. 

Re-tagging such files adding/removing artwork changes the behavior vice versa. 

An Apple lossless file without integrated artwork works well.

foobar2000, Picard, and exiftool display bits per sample for any of the above files in question. 

Can anybody confirm this behavior?

**Comment by KarlStraussberger** [comment]
Now, I understood your issue, found the source and created a new issue to track its fix

**Comment by ghlupe2** [comment]
Your latest fix does solve the issue on my side as well, thanks!

**Pull request merged**